File: PASCAL/pascal/d016 一元二次方程的实数根/d016 一元二次方程的实数根.pas
var
x1,x2,a,b,c,d:real;
begin
readln(a,b,c);
if b*b-4*a*c<0 then writeln('No answer!')
 else if b*b-4*a*c=0 then
  begin
   x1:=(-1*b+sqrt(b*b-4*a*c))/2/a;
   writeln(x1:0:2);
  end
   else if b*b-4*a*c>0 then
    begin
     x1:=(-1*b+sqrt(b*b-4*a*c))/2/a;
     x2:=(-1*b-sqrt(b*b-4*a*c))/2/a;
     writeln(x1:0:2);
     writeln(x2:0:2);
    end;
end.

File: CPP/SZNOI 语法百题/d016 一元二次方程的实数根/d016 一元二次方程的实数根.cpp
#include <iostream>
#include <math.h>
#include <iomanip>
using namespace std;
int main()
{
	float a, b, c;
	float m1, m2;
	cin >> a >> b >> c;
	if ((b*b - 4 * a*c)<0)
	{
		cout << "No answer£¡";
	}
	else
	{
		m1 = (-b + sqrt(b*b - 4 * a*c)) / (2 * a);
		m2 = (-b - sqrt(b*b - 4 * a*c)) / (2 * a);
		if (m1 == m2)
		{
			cout << setprecision(2) << setiosflags(ios::fixed) << m1;
		}
		else
		{
			cout << setprecision(2) << setiosflags(ios::fixed) << m1;
			cout << setprecision(2) << setiosflags(ios::fixed) << m2;
		}
	}
	return 0;
}